src/twisted/newsfragments/9151.bugfix

```diff
@@ -0,0 +1 @@
+twisted.internet.endpoints.HostnameEndpoint and twisted.internet.endpoints.TCP4Client endpoint documentation updated to correctly reflect that the timeout argument takes a float as well as an int.
```

Comment:
I think that this file should be ``.doc`` since there are no functional changes here.